### ISSUE TITLE
Stuck on deleting longhorn

### ISSUE THREAD

**Issue by senggen** [rancher/longhorn#344]
There is an "Waiting on external-attacher/io-rancher-longhorn" error, while delete pvc from rancher storage/persistant volume.

Tried commands below, the pvc and longhorn-system namespace is still there.
```sh
kubectl -n longhorn-system delete volumes.longhorn.rancher.io --all
kubectl -n longhorn-system delete engines.longhorn.rancher.io --all
kubectl -n longhorn-system delete replicas.longhorn.rancher.io --all
kubectl -n longhorn-system delete engineimages.longhorn.rancher.io --all
kubectl -n longhorn-system delete settings.longhorn.rancher.io --all
kubectl -n longhorn-system delete nodes.longhorn.rancher.io --all

kubectl -n longhorn-system delete deployment.apps/longhorn-driver-deployer
kubectl -n longhorn-system delete daemonset.apps/longhorn-csi-plugin
kubectl -n longhorn-system delete statefulset.apps/csi-attacher
kubectl -n longhorn-system delete service/csi-attacher
kubectl -n longhorn-system delete statefulset.apps/csi-provisioner
kubectl -n longhorn-system delete service/csi-provisioner
kubectl -n longhorn-system delete daemonset.apps/longhorn-flexvolume-driver

kubectl -n longhorn-system get daemonset.apps -o yaml | kubectl delete -f -
kubectl -n longhorn-system get deployment.apps -o yaml | kubectl delete -f -
kubectl -n longhorn-system get replicaset.apps -o yaml | kubectl delete -f -
kubectl -n longhorn-system get statefulset.apps -o yaml | kubectl delete -f -
kubectl -n longhorn-system get pods -o yaml | kubectl delete -f -
kubectl -n longhorn-system get service -o yaml | kubectl delete -f -

kubectl delete crd/engineimages.longhorn.rancher.io
kubectl delete crd/engines.longhorn.rancher.io
kubectl delete crd/nodes.longhorn.rancher.io
kubectl delete crd/replicas.longhorn.rancher.io
kubectl delete crd/volumes.longhorn.rancher.io
```

**Comment by yasker** [comment]
@senggen PVC needs to be removed before the Longhorn system was removed. https://github.com/rancher/longhorn#uninstall-longhorn

As a workaround, you can remove the `finalizer` in your pvc to get it removed.

**Comment by senggen** [comment]
@yasker Thanks for your answer.

I can't reinstall the longhorn, the error is "Validation failed in API: target namespace longhorn-system is not assigned to the current project c-xd6f2:p-lz8ml". And the "longhorn-system" in namespace is in removing status. 

So, I can't reinstall longhorn, and I can't remove it now.

I want to reinstall the longhorn, have any idea?

**Comment by yasker** [comment]
One possibility is the uninstallation of Longhorn wasn't done properly before, which left finalizer in the resources that cannot be cleaned up.

You can try the instructions at https://github.com/rancher/longhorn/blob/master/docs/chart.md#problems-with-crds to see if it helps to remove the Longhorn system namespace.

**Comment by senggen** [comment]
@yasker It's ok now, thank a lot.

**Comment by josecelano** [comment]
Same problem. I have created a new issue so people can find the issue faster:

https://github.com/longhorn/longhorn/issues/722

I have not found a solution yet.

Related issue:
https://github.com/longhorn/longhorn/issues/259
